src/net/socket.rs

```diff
@@ -65,26 +70,36 @@ impl Socket {
     pub fn start_polling(&mut self) -> Result<()> {
         // Nothing should break out of this loop!
         loop {
-            // First we pull any newly arrived packets and handle them
-            if let Err(e) = self.recv_from() {
-                error!("Encountered an error receiving data: {:?}", e);
-            };
-
-            // Now grab all the packets waiting to be sent and send them
-            while let Ok(p) = self.packet_receiver.try_recv() {
-                if let Err(e) = self.send_to(p) {
-                    match e {
-                        ErrorKind::IOError(ref e) if e.kind() == io::ErrorKind::WouldBlock => {}
-                        _ => error!("There was an error sending packet: {:?}", e),
-                    }
-                }
+            self.manual_poll()?;
+        }
+    }
+
+    /// Process any inbound/outbound packets and handle idle clients
+    pub fn manual_poll(&mut self) -> Result<()> {
```

Comment:
The way this is currently set up, you'll never return an Err result so you should probably remove the result type.